### PR TITLE
refactor: update with the latest protobuf and add unit test

### LINKED ISSUE #30: Add unittest

### LINKED ISSUE #33: Adopt gRPC status code
gRPC uses a set of [well-defined status codes](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md) for RPC API. We should adopt it.

### LINKED ISSUE #44: Adopt new protobuf
Related PR https://github.com/instill-ai/protobufs/pull/34

### PR DESCRIPTION
**Pull request by Phelan164**
Because

- Update with the protobuf change
- Support unit test

This commit

- Update with the latest protobuf which simplify output and update patch
- Support unit test
- close #44 
- close #30 
- close #33

### CODE REVIEW

**Review by pinglin — COMMENTED** [review]
Some minor comments when looking through the codes. The API change impacts the holistic codebase in terms of naming.

File: pkg/models/model.go [hunk 1]
```diff
@@ -123,14 +123,14 @@ type VersionResponse struct {
...
 	// Model name
 	Name string
 
 	FullName string
 
-	CVTask string
+	Task string
 
 	Versions []VersionResponse
 }
```
Comment (pinglin) [inline pkg/models/model.go:136]: The JSON response is not corresponding to the OpenAPI format:
```
{
  "model": {
    "id": "string",
    "name": "string",
    "fullName": "string",
    "task": "TASK_UNSPECIFIED",
    "modelVersions": [
      {
        "version": "string",
        "modelId": "string",
        "description": "string",
        "createdAt": "2022-03-20T19:38:16.387Z",
        "updatedAt": "2022-03-20T19:38:16.387Z",
        "status": "STATUS_UNSPECIFIED"
      }
    ]
  }
}
```

Comment (Phelan164) [reply pkg/models/model.go:136]: @pinglin I got your point. Yes, should be use the same struct

**Review by pinglin — APPROVED** [review]
LGTM

**Pull request merged**